modules/renter/persist.go

```diff
@@ -230,24 +241,40 @@ func (r *Renter) load() error {
 		}
 		return nil
 	})
-	if err != nil {
-		return err
+}
+
+// load fetches the saved renter data from disk.
+func (r *Renter) loadSettings() error {
+	r.persist = persistence{
+		Tracking: make(map[string]trackedFile),
+	}
+	err := persist.LoadJSON(settingsMetadata, &r.persist, filepath.Join(r.persistDir, PersistFilename))
+	if os.IsNotExist(err) {
+		// No persistence yet, set the defaults and continue.
+		r.persist.MaxDownloadSpeed = DefaultMaxDownloadSpeed
+		r.persist.MaxUploadSpeed = DefaultMaxUploadSpeed
+		r.persist.StreamCacheSize = DefaultStreamCacheSize
+	} else if err == persist.ErrBadVersion {
+		// Outdated version, try the 040 to 133 upgrade.
+		err = r.updatePersistVersionFrom040To133()
+		if err != nil {
+			// Nothing left to try.
+			return err
+		}
+		// Re-load the settings now that the file has been upgraded.
+		return r.loadSettings()
 	}
 
```

Comment:
Shouldn't we still be checking for `err != nil` after those two error conditions?